src/sentry/api/serializers/rest_framework/group_notes.py

```diff
@@ -1,7 +1,25 @@
 from __future__ import absolute_import
 
 from rest_framework import serializers
+from sentry.models import Project, User
+
+from .list import ListField
 
 
 class NoteSerializer(serializers.Serializer):
     text = serializers.CharField()
+    mentions = ListField(required=False)
+
+    def validate_mentions(self, attrs, source):
+        if source in attrs and 'group' in self.context:
+            mentions = attrs[source]
+            project = Project.objects.get(id=self.context['group'].project_id)
+        else:
+            mentions = None
+        if mentions:
+            users = User.objects.filter(id__in=mentions)
+            for u in users:
+                if not project.member_set.filter(user=u).exists():
+                    raise serializers.ValidationError('Cannot assign to non-team member')
```

Comment:
also, this block starting with `users = ...` could be reduced to a single query by doing something like:
```python
member_ids = set(project.member_set.filter(
    user_id__in=mentions,
).values_list('user_id', flat=True))
invalid_user_ids = [m for m in mentions if m not in member_ids]
if invalid_user_ids:
    raise ...
```